openhtf/io/output/print_console.py

```diff
@@ -0,0 +1,81 @@
+"""Module for outputting test record to JSON-formatted files."""
+import os
+import sys
+from json import JSONEncoder
+from openhtf import util
+
+
+class PrintConsole(JSONEncoder):
+  """Return an output callback that writes JSON Test Records.
+
+  An example filename_pattern might be:
+    '/data/test_records/%(dut_id)s.%(start_time_millis)s'
+
+  To use this output mechanism:
+    test = openhtf.Test(PhaseOne, PhaseTwo)
+    test.AddOutputCallback(openhtf.OutputToJson(
+        '/data/test_records/%(dut_id)s.%(start_time_millis)s'))
+
+  Args:
+    filename_pattern: A format string specifying the filename to write to,
+      will be formatted with the Test Record as a dictionary.
+    inline_attachments: Whether attachments should be included inline in the
+      output.  Set to False if you expect to have large binary attachments.
+  """
+
+  def __init__(self, filename_pattern=None, inline_attachments=True, **kwargs):
+    super(PrintConsole, self).__init__(**kwargs)
+    self.FAIL =  '\033[91m'
+    self.PASS =  '\033[92m'
```

Comment:
Agreed, I'd actually prefer to see these named by color instead of function, and see the function->color mapping explicitly elsewhere.

Also, it's better to define all-caps constants like this on the class rather than initializing them in __init__(), so you'll have something like:
```
class PrintConsole(object):
  GREEN = '\033[92m'
  RED = '\033[93m'
  ...
```